index.js

```diff
@@ -1,85 +1,150 @@
-  // don't allow digging while pathfinding
-  bot.allowDigWhilePathing(false);
+  // Disable rg-bot debug logging.  You can enable this to see more details about rg-bot api calls
+  bot.setDebug(false)
+
+  // Allow parkour so that our bots pathfinding will jump short walls and optimize their path for sprint jumps.
+  bot.allowParkour(true)
+
+  // We recommend disabling this on as you can't dig the CTF map.  Turning this on can lead pathfinding to get stuck.
+  bot.allowDigWhilePathing(false)
+
+  // Setup the rg-ctf-utils with debug logging
+  rgCtfUtils = new RGCTFUtils(bot)
+  rgCtfUtils.setDebug(true)
+
+  // Load the armor-manager plugin (https://github.com/PrismarineJS/MineflayerArmorManager)
+  bot.mineflayer().loadPlugin(armorManager)
+
+  // Information about the unbreakable block types
+  unbreakable = getUnbreakableBlockIds(bot)
+  console.log(`Unbreakable blocks: ${JSON.stringify(unbreakable)}`)
+
+}
+
+/**
+ * @param {RGBot} bot The configurable RGBot
+ */
+export async function runTurn(bot) {
+
+  try {
+
+    // find out which team I'm on
+    const myTeamName = bot.getMyTeam()
+    const otherTeamName = bot.matchInfo().teams.find(t => t.name !== myTeamName)?.name
 
-  // Instantiate our helper utilities and events for Capture the Flag
-  const rgctfUtils = new RGCTFUtils(bot);
+    // get my current position and log information about my state
+    const myPosition = bot.position()
+    console.log(`My team: ${myTeamName}, my position: ${bot.vecToString(myPosition)}, my inventory: ${JSON.stringify(bot.getAllInventoryItems().map((item) => nameForItem(item)))}`)
 
-  // track how many times we've died
-  let deaths = 0
+    // find any opponents in range
 
-  bot.on('death', () => {
-    console.log("I have died...")
-    ++deaths;
-    try {
-      // stop any current pathfinding goal
-      // @ts-ignore
-      bot.mineflayer().pathfinder.setGoal(null)
-      // @ts-ignore
-      bot.mineflayer().pathfinder.stop()
-    } catch (ex) {
+    const opponentNames = bot.getOpponentUsernames()
+    const opponents = bot.findEntities({
+      // opNames can be empty in practice mode where there is no other team
+      // if we don't pass some array to match, then this will return all entities instead
+      entityNames: (opponentNames.length === 0 && ['...']) || opponentNames,
+      attackable: true,
+      maxCount: 3,
+      maxDistance: 33, // Bots can only see ~30 +/1 blocks, so no need to search far
+      // override the default value function here as we aren't using this value in the sortValueFunction
+      entityValueFunction: (entityName) => {
+        return 0
+      },
+      // just sort them by distance for now... We'll filter them by decision point later
+      sortValueFunction: (distance, entityValue, health = 0, defense = 0, toughness = 0) => {
+        return distance
+      }
+    }).map(fr => fr.result)
 
+    // find any teammates in range
+    const teamMates = nearestTeammates(bot, 33, true)
```

Comment:
Not sure how much we care, but personally in my javascript bots, I have `type` jsdoc above each declared constant/variable so that my IDE does autocomplete.  This would definitely be 'nice' for users to have in our templates.

```javascript
    /**
     *  @type {Entity[]}
     */

```

Reply:
I did this in some places where the IDE was getting confused about the type, but didn't know how far we wanted to take it for the sample. I'll add some more of these 👍

Reply:
Done